tools/kmod.pc.in

```diff
@@ -4,6 +4,7 @@ distconfdir=@distconfdir@
 module_directory=@module_directory@
 module_compressions=@module_compressions@
 module_signatures=@module_signatures@
+depmod_module_directory_override=true
```

Comment:
humn... I think this favors too much the approach "query then do" rather than "do than fallback". It is expected that anyone mixing module_directory will have a recent kmod and not having it can be basically be checked on the error from passing --moduledir.

Reply:
this single patch can be backported to kmod 33 as a "fix" so it's available everywhere.

Reply:
The 'do else fallback' does not really work well here: depmod fails, and there is no 'fix'.

Reply:
```
kmod $ depmod --moduledir /tmp/inst
depmod: unrecognized option '--moduledir'
```

but there's no fallback if you query first, is there? Fallback here would be a proper error message to the user saying mixing module_directory is not supported by his depmod.   I think we may need to give it a try on the kernel patch and also ask Masahiro what what approach he prefers so we fix this the way that can be merged.